rest/view_api_test.go

```diff
@@ -620,3 +620,34 @@ func TestPostInstallCleanup(t *testing.T) {
 	goassert.Equals(t, len(postUpgradeResponse.Result["db"].RemovedDDocs), 0)
 
 }
+
+func TestViewQueryWrappers(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	rt.ServerContext().Database("db").SetUserViewsEnabled(true)
+
+	response := rt.SendAdminRequest("PUT", "/db/admindoc", `{"value":"foo"}`)
+	assertStatus(t, response, http.StatusCreated)
+	response = rt.SendAdminRequest("PUT", "/db/admindoc2", `{"value":"foo"}`)
+	assertStatus(t, response, http.StatusCreated)
+	response = rt.SendAdminRequest("PUT", "/db/userdoc", `{"value":"foo", "channels": ["userchannel"]}`)
+	assertStatus(t, response, http.StatusCreated)
+
+	response = rt.SendAdminRequest("PUT", "/db/_design/foodoc", `{"views": {"foobarview": {"map": "function(doc, meta) {if (doc.value == \"foo\") {emit(doc.key, null);}}"}}}`)
+	assert.Equal(t, 201, response.Code)
+
+	result, err := rt.WaitForNAdminViewResults(3, "/db/_design/foodoc/_view/foobarview")
+	assert.NoError(t, err)
+	assert.Equal(t, 3, result.Len())
+
+	a := rt.ServerContext().Database("db").Authenticator()
+	testUser, err := a.NewUser("testUser", "password", channels.SetOf("userchannel"))
+	assert.NoError(t, err)
+	err = a.Save(testUser)
+	assert.NoError(t, err)
+
+	result, err = rt.WaitForNUserViewResults(1, "/db/_design/foodoc/_view/foobarview", testUser, "password")
+	assert.NoError(t, err)
+	assert.Equal(t, 1, result.Len())
+}
```

Comment:
We should add a test that returns query results for both xattr and non-xattr documents in a single response, since that was the original motivation for this change.